### PR TITLE
Simplify publisher pipeline

### PR DESCRIPTION
**Pull request by urso**
- move event preprocessor normalizing packetbeat IPs and GeoIP support to packetbeat
- remove workers in sync/async publisher client modes directly forwarding events to output workers

### CODE REVIEW

File: libbeat/beat/beat.go [hunk 1]
```diff
@@ -56,12 +56,13 @@ type FlagsHandler interface {
...
-	Version string
-	Config  *BeatConfig
-	BT      Beater
-	Events  publisher.Client
-	UUID    uuid.UUID
+	Name      string
+	Version   string
+	Config    *BeatConfig
+	BT        Beater
+	Publisher *publisher.PublisherType
```
Comment (ruflin) [inline libbeat/beat/beat.go]: Naming makes now much more sense.

File: libbeat/beat/beat.go [hunk 2]
```diff
@@ -56,12 +56,13 @@ type FlagsHandler interface {
...
+	Events    publisher.Client
```
Comment (ruflin) [inline libbeat/beat/beat.go]: We should probably rename this to Client?

Comment (urso) [inline libbeat/beat/beat.go]: didn't want to change too much ;)

Comment (ruflin) [inline libbeat/beat/beat.go]: :+1:

File: filebeat/input/file.go [hunk 3]
```diff
@@ -72,6 +72,7 @@ func (f *FileEvent) ToMapStr() common.MapStr {
 		"message":    f.Text,
 		"type":       f.DocumentType,
 		"input_type": f.InputType,
+		"count":      1,
```
Comment (ruflin) [inline filebeat/input/file.go]: For reference: https://github.com/elastic/beats/issues/778

File: packetbeat/beat/packetbeat.go [hunk 4]
```diff
@@ -144,9 +146,16 @@ func (pb *Packetbeat) Setup(b *beat.Beat) error {
 
 	pb.Sniff = new(sniffer.SnifferSetup)
 
+	queueSize := 2048
```
Comment (ruflin) [inline packetbeat/beat/packetbeat.go]: Should this be a const?

File: packetbeat/tests/system/test_0013_redis_basic.py [hunk 5]
```diff
@@ -1,5 +1,7 @@
 from pbtests.packetbeat import TestCase
 
+import unittest
```
Comment (ruflin) [inline packetbeat/tests/system/test_0013_redis_basic.py]: How is this change related to the other changes?

**Comment by ruflin** [comment]
LGTM

File: packetbeat/beat/packetbeat.go (hunk 4 above)
Comment (urso) [inline packetbeat/beat/packetbeat.go]: yep

File: packetbeat/tests/system/test_0013_redis_basic.py (hunk 5 above)
Comment (urso) [inline packetbeat/tests/system/test_0013_redis_basic.py]: ups, leftover from testing.

**Comment by urso** [comment]
@ruflin squashed
